core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -157,7 +159,14 @@ public InspectContainerResponse getContainerInfo() {
                     while (true) {
                         RYUK_ACK_RATE_LIMITER.doWhenReady(() -> {
                             int index = 0;
-                            try(Socket clientSocket = new Socket(host, ryukPort)) {
+                            try(Socket clientSocket = new Socket()) {
+                                int socketReadTimeOut = TestcontainersConfiguration.getInstance().getRyukTimeout() * 1000;
+                                int socketConnectTimeOut = 5 * 1000;
+                                clientSocket.setSoTimeout(socketReadTimeOut);
```

Comment:
perhaps we should not set the read timeout, as Ryuk would not send anything unless a new filter is submitted, meaning that we would get a timeout exception pretty quick